packages/cln_plugin/lib/src/cln_plugin_base.dart

```diff
@@ -1,57 +1,93 @@
 // TODO: Put public facing types in this file.
 
+import 'dart:collection';
 import 'dart:io';
 
 import 'package:cln_plugin/src/rpcmethods/abstractrpcmethod.dart';
-import 'package:cln_plugin/src/rpcmethods/manifest/getmanifest.dart';
+import 'package:cln_plugin/src/rpcmethods/manifest/types/option.dart';
+import 'package:cln_plugin/src/rpcmethods/manifest/types/feature.dart';
 
 import 'icln_plugin_base.dart';
 
-class Plugin implements ICLNPlugin {
-  // Data members
-  late ManifestMethod manifest;
-  /*
-  Map<String, Object> parameters;
-  bool _parametersReady;
-  JsonConverter converter;
-  CLightningPluginConfig configs;
-   */
-
-  Plugin() {
+class Plugin implements CLNPlugin {
+  /// List of methods exposed
+  late List<RPCMethod> rpcMethods;
+
+  /// List of Subscriptions
+  late List<String> subscriptions;
+
+  /// List of Options
+  late List<Option> options;
+
+  /// List of Hooks
+  late Set<String> hooks;
+
+  /// FeatureBits for announcements of featurebits in protocol
+  late Feature features;
+
+  /// Boolean to mark dynamic management of plugin
+  late bool dynamic = false;
+
+  /// Custom notifications map
+  late HashMap<String, Function> notifications;
+
+  Plugin({bool dynamic = false}) {
     // Data member initializations
-    manifest = ManifestMethod();
   }
 
-  bool isNotReady = true;
-
   @override
   void start() {
     // This function will communicate with the rpc.
-    // print("The plugin is started.");
     try {
       String? messageSocket;
       while ((messageSocket = stdin.readLineSync()) != null) {
         // Already checked is stdin is not null, why trim and check again??
         if (messageSocket!.trim().isEmpty) {
           continue;
         }
-        // TODO: Log messageSocket to stdout as a debug message
-        int deep = 0;
-
-        // Dart doesn't have a max 32 bit int, and casting a double would return errors for some reason, hence the hardcode value.
-        final int maxDeep = 1073741824;
-
-        // TODO: Error handling for JSON and max recursion depth errors
       }
     } catch (error, stacktrace) {
       print('$error:$stacktrace');
-      //TODO: log this error
     }
   }
 
+  void addFeature(String node, String channel, String init, String invoice) {
+    features =
+        Feature(node: node, channel: channel, init: init, invoice: invoice);
+  }
+
+  @override
+  void registerRPCMethods(List<RPCMethod> methods) {
+    rpcMethods.addAll(methods);
+  }
+
+  @override
+  void registerRPCMethod(RPCMethod method) {
+    rpcMethods.add(method);
+  }
```

Comment:
```suggestion
  @override
  void registerRPCMethod({required String name}) {
    rpcMethods.add(RPCMethod(name: name));
  }
```

for all the parametes